state/relationunit.go

```diff
@@ -482,7 +483,26 @@ func (ru *RelationUnit) SettingsAddress() (network.Address, error) {
 	if crossmodel, err := ru.relation.IsCrossModel(); err != nil {
 		return network.Address{}, errors.Trace(err)
 	} else if !crossmodel {
-		return unit.PrivateAddress()
+		space, err := unit.GetSpaceForBinding("")
```

Comment:
This feels big enough that we should probably pull it out into a separate helper function (possibly exported for direct testing.)

Also, this shouldn't be the default space (I think). I think it should be the space *for the binding on this end of the relation*.

for example, if you do:
  juju deploy app1 --space "bar endpt1=foo"

then
  juju relate app1:endpt1 app2
should expose 'foo' as the private address, while if you do:
  juju relate app1:endpt2 app3
that should expose 'bar' as the private address.